api/app/src/mappings/garmin/bloodPressure.ts

```diff
@@ -0,0 +1,126 @@
+    };
+    // group by calendar date
+    const userDataByDate = groupBy(userData, (v) =>
+      toISODate(v.measurementTimeInSeconds)
+    );
+    const mappedItems: (UserData<Biometrics> | undefined)[] = Object.keys(
+      userDataByDate
+    ).map((date) => {
+      const userDataOfDate: GarminBloodPressure[] = userDataByDate[date].filter(
+        (d) => d.systolic != null || d.diastolic != null
+      );
+      if (userDataOfDate.length < 1) return undefined;
+      const distolic = mapToDiastolicSamples(userDataOfDate);
+      const systolic = mapToSystolicSamples(userDataOfDate);
+      return {
+        user,
+        typedData: {
+          type,
+          data: {
+            metadata: { date, source: PROVIDER_GARMIN },
+            blood_pressure: {
+              ...(distolic
+                ? {
+                    diastolic_mm_Hg: distolic,
+                  }
+                : undefined),
+              ...(systolic
+                ? {
+                    systolic_mm_Hg: systolic,
+                  }
+                : undefined),
+            },
+          },
+        },
+      };
+    });
+    const definedItems: UserData<Biometrics>[] = mappedItems.filter(
+      (v: UserData<Biometrics> | undefined) => v != undefined
+    ) as UserData<Biometrics>[];
+    return definedItems;
+  });
+};
+
+type DiastolicBloodPressure = DeepNonNullable<
+  DeepRequired<
+    Pick<GarminBloodPressure, "diastolic" | "measurementTimeInSeconds">
+  >
+>;
+export const mapToDiastolicSamples = (
+  garminBloodPressure: GarminBloodPressure[]
+): BloodPressure["diastolic_mm_Hg"] => {
+  const bp: DiastolicBloodPressure[] = garminBloodPressure.filter(
+    (v) => v.diastolic != null
+  ) as DiastolicBloodPressure[];
+  if (bp.length < 1) return undefined;
+  return {
+    samples: bp.map((v) => ({
+      time: toISODateTime(v.measurementTimeInSeconds),
+      value: v.diastolic,
+    })),
+  };
+};
+
+type SystolicBloodPressure = DeepNonNullable<
+  DeepRequired<
+    Pick<GarminBloodPressure, "systolic" | "measurementTimeInSeconds">
+  >
+>;
+export const mapToSystolicSamples = (
+  garminBloodPressure: GarminBloodPressure[]
+): BloodPressure["systolic_mm_Hg"] => {
+  const bp: SystolicBloodPressure[] = garminBloodPressure.filter(
+    (v) => v.systolic != null
+  ) as SystolicBloodPressure[];
+  if (bp.length < 1) return undefined;
+  return {
+    samples: bp.map((v) => ({
+      time: toISODateTime(v.measurementTimeInSeconds),
+      value: v.systolic,
+    })),
+  };
+};
+
+export const garminBloodPressureSchema = z.object({
+  // calendarDate: garminTypes.date,
+  measurementTimeInSeconds: garminTypes.measurementTime,
+  // measurementTimeOffsetInSeconds: -21600, // we always use UTC
+  // sourceType: z.enum(["MANUAL", "DEVICE"]), // not being used
+  systolic: garminTypes.systolic.nullable().optional(),
+  diastolic: garminTypes.diastolic.nullable().optional(),
+  // pulse: 77, // should we map this to Biometrics.heart_rate.samples_bpm?
```

Comment:
If it's not too much work, might as well - note we'd have to merge it into the existing samples list if any samples exist, might add a bit of complexity

Reply:
Added here: https://github.com/metriport/metriport/pull/20/commits/7c76b4088c53abf6db18f80f4b4ce68a73d33e3d

No changes to the SDK data model, so no update needed on the samples/docs?